### PR TITLE
tm 1

### PR DESCRIPTION
**Pull request by Jiaqi-beep**
- Original NCL Script: https://www.ncl.ucar.edu/Applications/Scripts/tm_1.ncl
- Original NCL Plots: https://www.ncl.ucar.edu/Applications/Images/tm_1_1_lg.png, https://www.ncl.ucar.edu/Applications/Images/tm_1_2_lg.png
- Issue: #342 
tm1

### CODE REVIEW

File: Plots/XY/NCL_tm_1.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
+"""
+NCL_tm_1.py
+===============
```
Comment (anissa111) [inline Plots/XY/NCL_tm_1.py:3]:
```suggestion
NCL_tm_1.py
===========
```

File: Plots/XY/NCL_tm_1.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+Tickmark and Axis Manipulation example
```
Comment (anissa111) [inline Plots/XY/NCL_tm_1.py:4]:
```suggestion
```
This doesn't need to be right there

File: Plots/XY/NCL_tm_1.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+Note: This script is aimed at demonstrating the explicit handling of tick marks, their locations, labels, etc.;
+therefore, the use of geocat-viz convenience functions is minimized here to show such tick management functions directly
+throughout this script.
```
Comment (anissa111) [inline Plots/XY/NCL_tm_1.py:8]: I would suggest moving this to be the last part of the top docstring to be more consistent with previous example formatting.

File: Plots/XY/NCL_tm_1.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+import matplotlib.pyplot as plt
+from matplotlib.ticker import MultipleLocator, AutoMinorLocator, FixedLocator, FormatStrFormatter
+
+from geocat.viz import util as gvutil
+
+###############################################################################
+# Generate data:
+
+# Note that range() top value is not included in the returned array of values.
+
```
Comment (anissa111) [inline Plots/XY/NCL_tm_1.py:33]:
```suggestion
```

No new line needed here

**Review by anissa111 — CHANGES_REQUESTED** [review]
This is looking good so far. I've included some minor formatting comments below. My main comment is that generally, when an example includes more than one plot, they should be recreated separately in most cases. 

I would also suggest adding `plt.tight_layout()` before your `plt.show()` call in most cases, too. In this combined plot, you can see the tick labels from the top plot and the tick marks from the bottom look a little too close. `plt.tight_layout()` fixes a lot of issues like this automatically.

File: Plots/XY/NCL_tm_1.py (hunk 3 above)
Comment (Jiaqi-beep) [reply Plots/XY/NCL_tm_1.py:7]: Hmm. I copied the format of tm_2 that looks like this.

"""
NCL_tm_2.py
===============
Tickmark and Axis Manipulation example

Note: This script is aimed at demonstrating the explicit handling of tick marks, their locations, labels, etc.;
therefore, the use of geocat-viz convenience functions is minimized here to show such tick management functions directly
throughout this script.

This script illustrates the following concepts:
   - Explicitly setting tickmarks and labels on the bottom X axis
   - Setting the spacing for tickmarks
   - Setting the mininum/maximum value of the Y axis in an XY plot
   - Changing the width and height of a plot

See following URLs to see the reproduced NCL plot & script:
    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/tm_2.ncl
    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/tm_1_1_lg.png and https://www.ncl.ucar.edu/Applications/Images/tm_1_2_lg.png
"""

Comment (anissa111) [reply Plots/XY/NCL_tm_1.py:7]: Huh, you're right about that. I would still remove the "Tickmark and Axis Manipulation example" but if you want to keep the Note above concepts illustrated, that's fine with me.

**Review by anissa111 — APPROVED** [review]
This looks great!

File: Plots/XY/NCL_tm_1.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+ax.yaxis.set_minor_locator(MultipleLocator(0.5))
+
+# Line-plot data
+plt.plot(x_data, y_data, color='grey', linewidth=0.5)
+
+# Use geocat.viz.util convenience function to set axes parameters without calling several matplotlib functions
+# Set axes limits, and tick values on x-axes.
+gvutil.set_axes_limits_and_ticks(ax,
+                                 xlim=(min(x_data) - 1, max(x_data) + 1),
+                                 ylim=(-4.5, 4.5))
```
Comment (michaelavs) [inline Plots/XY/NCL_tm_1.py:124]:
```suggestion
                                 ylim=(-4.5, 4.5)
                                 yticks= [-4, -2, 0, 2, 4],
                                 yticklabels=['-4.0', '-2.0', '0.0',  '2.0', '4.0'])
```
Define the correct y tick labels to match NCL plot. This should also be implemented in the first plot.

**Review by michaelavs — APPROVED** [review]
Looks good to me!

**Review by anissa111 — APPROVED** [review]

**Pull request merged**